client/src/components/Modals/EditTeam/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { ReactElement, FC, useState } from 'react';
+import {
+  Button,
+  ButtonProps,
+  Form,
+  Icon,
+  Input,
+  Modal,
+  ModalProps,
+} from 'semantic-ui-react';
+import Swal from 'sweetalert2';
+
+import { ITeam } from 'ssw-common';
+
+import './styles.scss';
+import { useTeams } from '../../../contexts';
+import { createManyTeams, updateManyTeams } from '../../../api';
+import { useEffect } from 'react';
+
+// interface TeamModalProps extends ModalProps {}
+
+const TeamModalTrigger: FC<ButtonProps> = ({ ...rest }): ReactElement => (
+  <Button {...rest} icon>
+    <Icon name="pencil" />
+  </Button>
+);
+
+// const copy = [...teams.map((t) => ({ ...t }))];
+
+const TeamModal: FC<ModalProps> = ({ ...rest }): ReactElement => {
+  const { teams, fetchTeams } = useTeams();
+  const [isOpen, setIsOpen] = useState(false);
+  const [formValues, setFormValues] = useState<ITeam[]>([]);
+
+  const [clonedTeams, setClonedTeams] = useState<Partial<ITeam>[]>([]);
+
+  useEffect(() => {
+    fetchTeams();
+    setFormValues(teams);
+
+    const clone = teams.map((team) => ({
+      _id: team._id,
+      name: team.name,
+      color: team.color,
+      active: team.active,
+    }));
+
+    setClonedTeams([...clone]);
+    console.log(teams);
+  }, [isOpen]);
```

Comment:
Add proper items to dependency array and make sure this doesn't break any functionality.